src/expression.js

```diff
@@ -1706,6 +1706,224 @@ function caml_binaryen_ref_eq(wasm_mod, left, right) {
   return wasm_mod.ref.func(left, right);
 }
 
+// Exception handling operations
+
+//Provides: caml_binaryen_try_native
+//Requires: caml_jsstring_of_string, caml_list_to_js_array
+function caml_binaryen_try_native(wasm_mod, name, body, catch_tags, catch_bodies, delegate_target) {
+  return wasm_mod.try(
+    name ? caml_jsstring_of_string(name[1]) : null,
+    body,
+    caml_list_to_js_array(catch_tags).map(caml_jsstring_of_string),
+    caml_list_to_js_array(catch_bodies),
+    delegate_target ? caml_jsstring_of_string(delegate_target[1]) : null,
+  );
+}
+
+//Provides: caml_binaryen_try_bytecode
+//Requires: caml_binaryen_try_native
+function caml_binaryen_try_bytecode() {
+  return caml_binaryen_try_native(arguments[0], arguments[1], arguments[2], arguments[3], arguments[4], arguments[5]);
+}
+
+//Provides: caml_binaryen_throw
+//Requires: caml_jsstring_of_string, caml_list_to_js_array
+function caml_binaryen_throw(wasm_mod, tag, operands) {
+  return wasm_mod.throw(
+    caml_jsstring_of_string(tag),
+    caml_list_to_js_array(operands),
+  );
+}
+
+//Provides: caml_binaryen_rethrow
+//Requires: caml_jsstring_of_string
+function caml_binaryen_rethrow(wasm_mod, target) {
+  return wasm_mod.rethrow(
+    caml_jsstring_of_string(target),
+  )
+}
+
+//Provides: caml_binaryen_try_get_name
+//Requires: Binaryen, caml_string_of_jsstring
+function caml_binaryen_try_get_name(expr) {
+  return caml_string_of_jsstring(Binaryen['_BinaryenTryGetName'](expr));
```

Comment:
Were you able to verify that this works? I don't know if this actually returns a JS string or if it's just a pointer or something.

Reply:
I don't think any of the raw `_Binaryen` functions actually return strings, as those would be operating in Asm.js memory space